include/public/marco/Codegen/Lowering/VariablesSymbolTable.h

```diff
@@ -0,0 +1,38 @@
+#ifndef MARCO_CODEGEN_LOWERING_VARIABLESSYMBOLTABLE_H
+#define MARCO_CODEGEN_LOWERING_VARIABLESSYMBOLTABLE_H
+
+#include "marco/Codegen/Lowering/Reference.h"
+#include "llvm/ADT/ScopedHashTable.h"
+#include <optional>
+#include <set>
+#include <string>
+
+namespace marco::codegen::lowering
+{
+  class VariablesSymbolTable {
+    public:
+      class VariablesScope {
```

Comment:
Where is the destructor in charge of removing the variables when exiting the scope?

Reply:
If you are referring to removing the variables from `references` (formerly called `hashTable`), than, as far as I know, when `VariablesScope` is destroyed, its members' (including `scope`) destructors are called, removing the variables.
If you are instead referring to removing the variables from `variablesSet`, this is not done since it cannot be done for the reason I stated under the conversation on `ForEquationLowerer.cpp`. If case the variables need to be removed, a different solution has to be found.

Reply:
I was referring to the latter case. However with the current design this may be acceptable.